cli/ops/native_plugins.rs

```diff
@@ -0,0 +1,99 @@
+use super::dispatch_json::{Deserialize, JsonOp, Value};
+use crate::fs as deno_fs;
+use crate::ops::json_op;
+use crate::state::ThreadSafeState;
+use deno::*;
+use dlopen::symbor::Library;
+use std::collections::HashMap;
+use std::ffi::OsStr;
+use std::sync::Arc;
+
+pub fn init(i: &mut Isolate, s: &ThreadSafeState, r: Arc<deno::OpRegistry>) {
+  let r_ = r.clone();
+  i.register_op(
+    "open_native_plugin",
+    s.core_op(json_op(s.stateful_op(move |state, args, zero_copy| {
+      op_open_native_plugin(&r_, state, args, zero_copy)
+    }))),
+  );
+}
+
+fn open_plugin<P: AsRef<OsStr>>(lib_path: P) -> Result<Library, ErrBox> {
+  debug!("Loading Native Plugin: {:#?}", lib_path.as_ref());
+
+  Library::open(lib_path).map_err(ErrBox::from)
+}
+
+struct NativePluginResource {
+  lib: Library,
+  ops: Vec<(String, OpId)>,
+}
+
+impl Resource for NativePluginResource {}
+
+struct InitContext {
+  registry: Arc<deno::OpRegistry>,
+  plugin_rid: ResourceId,
+  ops: Vec<(String, OpId)>,
+}
+
+impl PluginInitContext for InitContext {
+  fn register_op(
+    &mut self,
+    name: &str,
+    op: Box<dyn Fn(&[u8], Option<PinnedBuf>) -> CoreOp + Send + Sync + 'static>,
+  ) -> OpId {
+    let opid = self
```

Comment:
My previous comment still stands - I think actual registration should happen in `op_open_native_plugin` after call to `init_fn`.  `registry` and `plugin_rid` shouldn't be exposed to plugin itself

Reply:
I might rework this a little, but I want to get tests working soon.